src/controller/python/chip/ChipDeviceCtrl.py

```diff
@@ -484,7 +484,83 @@ async def ReadAttribute(self, nodeid: int, attributes: typing.List[typing.Union[
+        typing.Tuple[typing.Type[ClusterObjects.ClusterEventDescriptor]],
+        # Wildcard event id
+        typing.Tuple[int, typing.Type[ClusterObjects.Cluster]],
+        # Concrete path
+        typing.Tuple[int, typing.Type[ClusterObjects.ClusterEventDescriptor]]
+    ]], reportInterval: typing.Tuple[int, int] = None):
+        '''
+        Read a list of events from a target node
+
+        nodeId: Target's Node ID
+        events: A list of tuples of varying types depending on the type of read being requested:
+            (endpoint, Clusters.ClusterA.EventA):       Endpoint = specific,    Cluster = specific,   Event = specific
+            (endpoint, Clusters.ClusterA):              Endpoint = specific,    Cluster = specific,   Event = *
+            (Clusters.ClusterA.EventA):                 Endpoint = *,           Cluster = specific,   Event = specific
+            endpoint:                                   Endpoint = specific,    Cluster = *,          Event = *
+            Clusters.ClusterA:                          Endpoint = *,           Cluster = specific,   Event = *
+            '*' or ():                                  Endpoint = *,           Cluster = *,          Event = *
+
+        The cluster and events specified above are to be selected from the generated cluster objects.
+
+        e.g.
+            ReadEvent(1, [ 1 ] ) -- case 4 above.
+            ReadEvent(1, [ Clusters.Basic ] ) -- case 5 above.
+            ReadEvent(1, [ (1, Clusters.Basic.Events.Location ] ) -- case 1 above.
+
+        reportInterval: A tuple of two int-s for (MinIntervalFloor, MaxIntervalCeiling). Used by establishing subscriptions.
+            When not provided, a read request will be sent.
+        '''
+
+        eventLoop = asyncio.get_running_loop()
+        future = eventLoop.create_future()
+
+        device = self.GetConnectedDeviceSync(nodeid)
+        eves = []
+        for v in events:
+            endpoint = None
+            cluster = None
+            event = None
+            if v == ('*') or v == ():
+                # Wildcard
+                pass
+            elif type(v) is not tuple:
+                print(type(v))
+                if type(v) is int:
+                    endpoint = v
+                elif issubclass(v, ClusterObjects.Cluster):
+                    cluster = v
+                elif issubclass(v, ClusterObjects.ClusterEventDescriptor):
+                    event = v
+                else:
+                    raise ValueError("Unsupported Event Path")
+            else:
+                # endpoint + (cluster) event / endpoint + cluster
+                endpoint = v[0]
+                if issubclass(v[1], ClusterObjects.Cluster):
+                    cluster = v[1]
+                elif issubclass(v[1], ClusterAttribute.ClusterEventDescriptor):
+                    event = v[1]
+                else:
+                    raise ValueError("Unsupported Attribute Path")
```

Comment:
What about using kind of `switch` approach?
```suggestion
            print(type(v))
            if type(v) is tuple:
                # not a wildcard
                if not v in [('*'), ()]:
                    # endpoint + (cluster) event / endpoint + cluster
                    endpoint = v[0]
                    if issubclass(v[1], ClusterObjects.Cluster):
                        cluster = v[1]
                    elif issubclass(v[1], ClusterAttribute.ClusterEventDescriptor):
                        event = v[1]
                    else:
                        raise ValueError("Unsupported Attribute Path")
            elif type(v) is int:
                endpoint = v
            elif issubclass(v, ClusterObjects.Cluster):
                cluster = v
            elif issubclass(v, ClusterObjects.ClusterEventDescriptor):
                event = v
            else:
                 raise ValueError("Unsupported Event Path")
```

Reply:
this logic is similar to what we use for attribute, and would update it in follow-up PR with attribute together.